mods/letsgo/scripts.js

```diff
@@ -47,7 +47,7 @@ let BattleScripts = {
 		if (nature.plus) stats[nature.plus] = Math.floor(stats[nature.plus] * 1.1);
 		// @ts-ignore
 		if (nature.minus) stats[nature.minus] = Math.floor(stats[nature.minus] * 0.9);
-		let friendshipValue = Math.floor((this.clampIntRange(set.happiness || 255, 0, 255) / 255 / 10 + 1) * 100);
+		let friendshipValue = Math.floor((set.happiness / 255 / 10 + 1) * 100);
```

Comment:
the error is caused because you removed the ``|| 255`` from this, so there's no check for a falsy value